neuralpp/symbolic/interval.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+from abc import abstractmethod
+from typing import Iterable, Callable
+from constants import if_then_else
+
+from .expression import Variable, Expression, Context
+from .z3_expression import Z3SolverExpression
+
+
+class ClosedInterval:
```

Comment:
I think this should also inherit `GeneralIntervals', since this method also needs `from_constraint`.

Reply:
The class `GeneralIntervals` is gone in the new commit